portal-impl/src/com/liferay/portal/tools/HypersonicLoader.java

```diff
@@ -95,10 +95,13 @@ public HypersonicLoader(
 		// See LEP-2927. Appending ;shutdown=true to the database connection URL
 		// guarantees that ${databaseName}.log is purged.
 
+		String connectionProperties =
+			";hsqldb.write_delay=false;shutdown=true;sql.syntax_ora=true";
+
 		try (Connection connection = DriverManager.getConnection(
 				StringBundler.concat(
 					"jdbc:hsqldb:", sqlDir, "/", databaseName,
-					";hsqldb.write_delay=false;shutdown=true"),
+					connectionProperties),
 				userName, password)) {
 
 			if (Validator.isNull(fileNames)) {
```

Comment:
@thienquach87, the fix looks unrelated with the second commit. Since HypersonicLoader class isn't loaded during portal runtime. If the fix can take effect, why not add the fix to the property **jdbc.default.url** in portal.properties?

Reply:
You are right @yuhai, we also add jdbc.default.url=jdbc:hsqldb:${liferay.home}/data/hypersonic/lportal;sql.syntax_ora=true to add the sql.syntax_ora property to 'true' value. However if we do it in this way, it is necessary to take note it to end user.
Finally, I updated source following your comment. Please help review it again and leave your feed-back. 

Thank you!